.github/workflows/test_deploy.yml

```diff
@@ -558,18 +558,21 @@ jobs:
         name: release-url
         path: release-url.txt
 
-  push-image:
+  build-images:
     runs-on: ubuntu-latest
-    needs: [docs-linux, test-linux-cli, test-linux-core, test-linux-service, test-macos-cli, test-macos-core, test-macos-service]
-    if: "startsWith(github.ref, 'refs/tags/') || github.ref == 'refs/heads/master'"
+    # needs: [docs-linux, test-linux-cli, test-linux-core, test-linux-service, test-macos-cli, test-macos-core, test-macos-service]
     steps:
     - uses: actions/checkout@v2
+    - name: build images
+      run: make docker-tag
     - name: Docker Login
+      if: "startsWith(github.ref, 'refs/tags/') || github.ref == 'refs/heads/master'"
       uses: Azure/docker-login@v1
       with:
         username: ${{ secrets.DOCKER_USERNAME }}
         password: ${{ secrets.DOCKER_PASSWORD }}
-    - name: build and push image
+    - name: push images
+      if: "startsWith(github.ref, 'refs/tags/') || github.ref == 'refs/heads/master'"
```

Comment:
same as above. i'm wondering if it would be possible to make this at least a workflow parameter, or actually check whether the github.ref == `git symbolic-ref refs/remotes/origin/HEAD`.
but this is just a really minor details

Reply:
good idea - I'd like to make this more robust... but won't your suggestion always be false?

Reply:
btw we use this pattern elsewhere - I'd say we should make an issue to fix it in general and maybe also propagate that fix to other component repos